include/swift/SIL/DebugUtils.h

```diff
@@ -218,6 +218,70 @@ inline SILBasicBlock::iterator eraseFromParentWithDebugInsts(
 bool hasNonTrivialNonDebugTransitiveUsers(
     PointerUnion<SILInstruction *, SILArgument *> V);
 
+/// A high level composition struct on top of debug value instructions that
+/// enable one to write abstract code over all debug insts.
+struct DebugInst {
+  enum class Kind {
+    Invalid = 0,
+    ObjectValue,
+    AddressValue,
```

Comment:
I think Value and Address would be more consistent with the usual terminology.

Reply:
I made it more specific in a new version I am finishing up real quick.